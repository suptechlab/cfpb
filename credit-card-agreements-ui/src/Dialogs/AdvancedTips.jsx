import './AdvancedTips.less';
import React from 'react';

export default class AdvancedTips extends React.Component {

  render() {
    return (
      <nav className="advanced_tips">
          <div className="o-expandable">
            <div className="o-expandable_content
              o-expandable_content__onload-open">
              <h3>Advanced search tips</h3>
              <div className="tips">
                <div className="tip">
                  <h4>And/Or</h4>
                  <input className="a-text-input example-input first"
                          type="text" readOnly value="Credit AND Card"/>
                  <input className="a-text-input example-input"
                          type="text" readOnly value="Loan OR Mortgage"/>
                  <div className="tip-description">
                    <p>
                      Use AND when results must contain both terms
                    </p>
                    <p>
                      Use OR when results should contain at least one of the
                      terms
                    </p>
                  </div>
                </div>
                <div className="tip">
                  <h4>Wildcard search</h4>
                  <input className="a-text-input example-input first"
                          type="text" readOnly value="pay*"/>
                  <input className="a-text-input example-input"
                          type="text" readOnly value="Mort?age"/>
                  <div className="tip-description">
                    <p>
                      Use * after a few letters or a word to
                       show results that start with those letters or word
                    </p>
                    <p>
                      Use ? to show results where the “?” is
                      replaced by any character
                    </p>
                  </div>
                </div>
                <div className="tip">
                  <h4>Must/must not contain</h4>
                  <input className="a-text-input example-input first"
                          type="text" readOnly value="+foreclosure"/>
                  <input className="a-text-input example-input"
                          type="text" readOnly
                          value="-&#34;debt collector&#34;" />
                  <div className="tip-description">
                    <p>
                      Use + if the search must contain the term
                    </p>
                    <p>
                      Use - if the search must not contain the term
                    </p>
                  </div>
                </div>
                <div className="tip">
                  <h4>Proximity Search</h4>
                  <input className="a-text-input example-input_full"
                          type="text" readOnly
                          value="&#34;Loan default&#34;~3"/>
                  <div className="tip-description">
                    <p>
                      Use ~[#] to search for two words within that number of
                      characters from one another; the words should be in
                      quotation marks
                    </p>
                  </div>
                </div>
                <div className="tip">
                  <h4>Fuzzy Search</h4>
                  <input className="a-text-input example-input_full"
                          type="text" readOnly value="Mortgage~2"/>
                  <div className="tip-description">
                    <p>
                      Use ~[#] to search for terms that are similar to your
                      keyword terms within a certain margin of error
                      (~1 catches 80% of spelling errors)
                    </p>
                  </div>
                </div>
                <div className="tip">
                  <h4>Boost Search</h4>
                  <input className="a-text-input example-input_full"
                          type="text" readOnly value="pay^2 bill"/>
                  <div className="tip-description">
                    <p>
                      Use ~2 after a term to boost its relevance above the
                      other term
                    </p>
                  </div>
                </div>
              </div>
            </div>
        </div>
      </nav>
    );
  }
}
