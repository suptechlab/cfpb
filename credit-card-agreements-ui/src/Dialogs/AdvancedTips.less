@import (less) "../base.less";

.advanced_tips {
  box-sizing: border-box;
  background-color: @gray-20;

  //override parent padding
  margin-left: -15px;
  margin-right: -15px;
  margin-bottom: -15px;
  margin-top: 15px;

  padding: 15px;

  .tips {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
  }
  .tip {
    flex-basis: 31%;
    padding-right: 2%;
    padding-top: 15px;
    padding-bottom: 15px;
    .example-input {
      box-sizing: border-box;
      width: 48%;
      color: @gray-80;
      display:inline-block;
      &.first {
        margin-right: 1%
      }
    }
    .example-input_full {
      box-sizing: border-box;
      color: @gray-80;
      width: 100%;
    }
    .tip-description {
      padding-top: 10px;
    }
  }

}
