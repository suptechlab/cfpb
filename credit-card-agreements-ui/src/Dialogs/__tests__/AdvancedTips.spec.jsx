import { shallow } from 'enzyme'
import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import AdvancedTips from '../AdvancedTips'

function setupSnapshot() {
  return renderer.create(
      <AdvancedTips />
  )
}

describe('component::AdvancedTips', () => {
  describe('initial state', () => {
    it('renders HTML', () => {
      const target = setupSnapshot()
      const tree = target.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
