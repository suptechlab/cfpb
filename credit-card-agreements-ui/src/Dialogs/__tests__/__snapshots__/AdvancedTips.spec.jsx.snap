// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component::AdvancedTips initial state renders HTML 1`] = `
<nav
  className="advanced_tips"
>
  <div
    className="o-expandable"
  >
    <div
      className="o-expandable_content o-expandable_content__onload-open"
    >
      <h3>
        Advanced search tips
      </h3>
      <div
        className="tips"
      >
        <div
          className="tip"
        >
          <h4>
            And/Or
          </h4>
          <input
            className="a-text-input example-input first"
            readOnly={true}
            type="text"
            value="Credit AND Card"
          />
          <input
            className="a-text-input example-input"
            readOnly={true}
            type="text"
            value="Loan OR Mortgage"
          />
          <div
            className="tip-description"
          >
            <p>
              Use AND when results must contain both terms
            </p>
            <p>
              Use OR when results should contain at least one of the terms
            </p>
          </div>
        </div>
        <div
          className="tip"
        >
          <h4>
            Wildcard search
          </h4>
          <input
            className="a-text-input example-input first"
            readOnly={true}
            type="text"
            value="pay*"
          />
          <input
            className="a-text-input example-input"
            readOnly={true}
            type="text"
            value="Mort?age"
          />
          <div
            className="tip-description"
          >
            <p>
              Use * after a few letters or a word to show results that start with those letters or word
            </p>
            <p>
              Use ? to show results where the “?” is replaced by any character
            </p>
          </div>
        </div>
        <div
          className="tip"
        >
          <h4>
            Must/must not contain
          </h4>
          <input
            className="a-text-input example-input first"
            readOnly={true}
            type="text"
            value="+foreclosure"
          />
          <input
            className="a-text-input example-input"
            readOnly={true}
            type="text"
            value="-\\"debt collector\\""
          />
          <div
            className="tip-description"
          >
            <p>
              Use + if the search must contain the term
            </p>
            <p>
              Use - if the search must not contain the term
            </p>
          </div>
        </div>
        <div
          className="tip"
        >
          <h4>
            Proximity Search
          </h4>
          <input
            className="a-text-input example-input_full"
            readOnly={true}
            type="text"
            value="\\"Loan default\\"~3"
          />
          <div
            className="tip-description"
          >
            <p>
              Use ~[#] to search for two words within that number of characters from one another; the words should be in quotation marks
            </p>
          </div>
        </div>
        <div
          className="tip"
        >
          <h4>
            Fuzzy Search
          </h4>
          <input
            className="a-text-input example-input_full"
            readOnly={true}
            type="text"
            value="Mortgage~2"
          />
          <div
            className="tip-description"
          >
            <p>
              Use ~[#] to search for terms that are similar to your keyword terms within a certain margin of error (~1 catches 80% of spelling errors)
            </p>
          </div>
        </div>
        <div
          className="tip"
        >
          <h4>
            Boost Search
          </h4>
          <input
            className="a-text-input example-input_full"
            readOnly={true}
            type="text"
            value="pay^2 bill"
          />
          <div
            className="tip-description"
          >
            <p>
              Use ~2 after a term to boost its relevance above the other term
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</nav>
`;
